prometheus/client.go

```diff
@@ -92,9 +92,9 @@ func (in *Client) GetSourceServices(namespace string, servicename string) (map[s
 	return routes, nil
 }
 
-// GetServiceMetrics returns the Metrics related to the provided service identified by its namespace and service name.
-func (in *Client) GetServiceMetrics(query *ServiceMetricsQuery) Metrics {
-	return getServiceMetrics(in.api, query)
+// GetMetrics returns the Metrics related to the provided service identified by its namespace and service name.
```

Comment:
Not sure if this comment is valid now... "related to the provided service identified by its namespace and service name".

Reply:
I guess this is sort of my question.  Whether we are even asking for "service" metrics anymore.  Requests are made from a workload to a service, but the destination is also a workload except in the case the that request is denied in the source-side proxy (i.e. 503).    So honestly, we're really querying for workload metrics, but "app" is the identifier.  And so I wonder whether the API should change to that terminology (i.e. Service -> App).  Incoming metrics would represent requests serviced by the workload/app.  Outgoing metrics would represent requests attempted, whether they were sent by the proxy or not.

So, not that a source does have an app and version labels.  If we eliminate version from the query then we would be aggregating across versions of the app.  I think the API will probably need to be able to fetch metrics for an app and app+version.

Reply:
We have source_app, source_version and source_workload.  I thought we would use the two values: source_app and source_version,  but you are saying we should use source_workload as the identifier for metrics?

Reply:
yes, source_workload replaces our current (source_app + source_version)
but source_app alone is still interesting when we need coarser grain.